src/main/java/com/fincons/service/authService/RoleServiceImpl.java

```diff
@@ -71,16 +87,52 @@ public RoleDTO updateRole(long roleId, RoleDTO roleModifiedDTO) throws  RoleExce
             throw new ResourceNotFoundException("Role does not Exist");
         }
 
+        if(roleFound.get().getName().equals("ROLE_ADMIN")){
+            throw new RoleException("Can't modify a ROLE_ADMIN");
+        }
+
         Role roleToModify = roleFound.get();
 
         roleToModify.setName(roleModifiedDTO.getName().toUpperCase());
-        Role roleModifiedSaved =  roleRepository.save(roleToModify);
 
-        return userAndRoleMapper.roleToRoleDto(roleModifiedSaved);
+        return roleRepository.save(roleToModify);
     }
 
+    @Override
+    public GenericResponse<ReturnObject> deleteRole(long roleId, boolean deleteUsersAssociated) throws RoleException {  //  if Boolean deleteUsersAssociated == null -- > nullPointerExepciont
```

Comment:
- ReturnObject is not needed, you should return a message in case of you properly deleted a role and users; otherwise you should build an exception message

-  boolean deleteUsersAssociated can't produce a nullPointerException